sdk/samples/core/mosquitto_rpc_client_sample.c

```diff
@@ -0,0 +1,303 @@
+
+  az_mqtt5 mqtt5;
+  struct mosquitto* mosq = NULL;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_init(&mqtt5, &mosq, NULL));
+
+  az_mqtt5_x509_client_certificate primary_credential = (az_mqtt5_x509_client_certificate){
+    .cert = cert_path1,
+    .key = key_path1,
+  };
+
+  az_mqtt5_connection_options connection_options = az_mqtt5_connection_options_default();
+  connection_options.client_id_buffer = client_id;
+  connection_options.username_buffer = username;
+  connection_options.password_buffer = AZ_SPAN_EMPTY;
+  connection_options.hostname = hostname;
+  connection_options.client_certificates[0] = primary_credential;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_connection_init(
+      &mqtt_connection, &connection_context, &mqtt5, mqtt_callback, &connection_options));
+
+  az_mqtt5_property_bag property_bag;
+  mosquitto_property* mosq_prop = NULL;
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_property_bag_init(&property_bag, &mqtt5, &mosq_prop));
+
+  LOG_AND_EXIT_IF_FAILED(az_rpc_client_policy_init(
+      &rpc_client_policy,
+      &rpc_client,
+      &mqtt_connection,
+      property_bag,
+      client_id,
+      model_id,
+      command_name,
+      AZ_SPAN_FROM_BUFFER(response_topic_buffer),
+      AZ_SPAN_FROM_BUFFER(request_topic_buffer),
+      AZ_SPAN_FROM_BUFFER(subscription_topic_buffer),
+      NULL));
+
+  LOG_AND_EXIT_IF_FAILED(pending_commands_array_init(&pending_commands, correlation_id_buffers));
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_connection_open(&mqtt_connection));
+
+  az_result rc;
+
+  // infinite execution loop
+  for (int i = 45; !sample_finished && i > 0; i++)
+  {
+    // remove any expired commands
+    pending_command* expired_command = get_first_expired_command(pending_commands);
+    while (expired_command != NULL)
+    {
+      printf(LOG_APP_ERROR "command ");
+      print_correlation_id(expired_command->correlation_id);
+      printf(" expired\n");
+      az_result ret = remove_command(&pending_commands, expired_command->correlation_id);
+      if (ret != AZ_OK)
+      {
+        printf(LOG_APP_ERROR "Command not found\n");
+      }
+      expired_command = get_first_expired_command(pending_commands);
+    }
+#ifdef _WIN32
+    Sleep((DWORD)1000);
+#else
+    sleep(1);
+#endif
+    printf(LOG_APP "Waiting...\r");
+    fflush(stdout);
+    // invokes a command every 15 seconds. This cadence/how it is triggered should be customized for
+    // your solution.
+    if (i % 15 == 0)
+    {
+      uuid_t new_uuid;
+      uuid_generate(new_uuid);
+      az_mqtt5_rpc_client_invoke_req_event_data command_data
+          = { .correlation_id
+              = az_span_create((uint8_t*)new_uuid, AZ_MQTT5_RPC_CORRELATION_ID_LENGTH),
+              .content_type = content_type,
+              .rpc_server_client_id = server_client_id,
+              .request_payload = AZ_SPAN_FROM_STR(
+                  "{\"RequestTimestamp\":1691530585198,\"RequestedFrom\":\"mobile-app\"}") };
```

Comment:
1691530585198, use get time?

Reply:
This is a hardcoded payload for now until we have json serialization, so I'll leave this as is with a comment that generating this is future work